src/app/icd/client/DefaultCheckInDelegate.cpp

```diff
@@ -44,5 +44,44 @@ void DefaultCheckInDelegate::OnCheckInComplete(const ICDClientInfo & clientInfo)
 #endif
 }
 
+ICDRefreshKeyInfo * DefaultCheckInDelegate::OnKeyRefreshNeeded(const ICDClientInfo & clientInfo,
+                                                               const ICDClientStorage * clientStorage)
+{
+    auto icdRefreshKeyInfo = CreateICDRefreshKeyInfo(clientInfo, clientStorage);
+
+    if (icdRefreshKeyInfo == nullptr)
+    {
+        return nullptr;
+    }
+
+    CHIP_ERROR err = chip::Crypto::DRBG_get_bytes(icdRefreshKeyInfo->mNewKey.Bytes(), icdRefreshKeyInfo->mNewKey.Capacity());
+    if (err != CHIP_NO_ERROR)
+    {
+        return nullptr;
+    }
+
+    return icdRefreshKeyInfo;
+}
+
+void DefaultCheckInDelegate::OnRegistrationUpdateComplete(const ICDClientInfo & clientInfo, ICDRefreshKeyInfo * icdRefreshKeyInfo)
+{
+    ChipLogProgress(ICD, "Re-registration with new key successful. New start counter = %d", clientInfo.start_icd_counter);
+    ReleaseICDRefreshKeyInfo(icdRefreshKeyInfo);
+}
+void DefaultCheckInDelegate::OnRegistrationUpdateFailure(ICDRefreshKeyInfo * icdRefreshKeyInfo)
+{
+    ReleaseICDRefreshKeyInfo(icdRefreshKeyInfo);
+}
+
+ICDRefreshKeyInfo * DefaultCheckInDelegate::CreateICDRefreshKeyInfo(const ICDClientInfo & clientInfo,
+                                                                    const ICDClientStorage * clientStorage)
+{
+    return chip::Platform::MakeUnique<ICDRefreshKeyInfo>(this, clientInfo, clientStorage).get();
+}
+void DefaultCheckInDelegate::ReleaseICDRefreshKeyInfo(ICDRefreshKeyInfo * icdRefreshKeyInfo)
+{
+    chip::Platform::Delete(icdRefreshKeyInfo);
```

Comment:
This pointer was not allocated with Platform::New, so is Platform::Delete the right thing?

Reply:
The pointer was allocated using Platform::New earlier and then I was changing things around. Didn't get to update everything but pushed into a PR to save my work. I should have marked the PR as "Not for review". That was my bad. Sorry you had wasted time on a draft PR.